### ISSUE TITLE
Identifier too long

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#2675]
Identifier too long
[Live Deck](https://deck.net/80b0a32653b8a0ffca416d43a1196cdc)
In .NET, identifiers can be as long as you want.

**Comment by vladsch** [comment]
Unfortunately, it is limitation of NRefactory C# parser

**Comment by Suchiman** [comment]
Actually, that is [not true](https://sharplab.io/#v2:CYLg1APgAgzABFATHAqgOwJYHs0H0BKAKgMIDKALgIbkDOhAngA4CmuGaARlgK5rABO5RrnTY8RMlVoMWIzDnwYA5gAtyuHuS68BQuWIIkK1Ok1aiFytfsur1NZjRoGL4o1NOzXiuzbQ/rbyt1YGpKAGMVSjQ0ZgAbP0NJExlzeX9gvwD1IN9c63J+CIBrRIljaTMszPyc9OzqvPrMwuiaRixBMvcUqtrGwOamsQb+/vDo4AxQ8mZGSgx+buTKryHBkZr1us3h2w39ncPcOKwJuIm+aepWco9UgaOMveeD16fR7cfv/mYAWyws0uUxmzFcSQqnlY7G0fEEwnBd16a121k0sN0CPSEPufS+/QcThc2KRqzSqI+WwpuBmESiMXiy0hDzG+LZ1Na4VKiJ6ZO+rOpAuOrTQ7U6TxxyPJAFgAFAAbzlAF8gA=).
The Limit in roslyn is 1024 characters, starting with 1025 characters you receive the following error:
`error CS7013: Name 'Union_[...]RTCStatsType_Uni' exceeds the maximum length allowed in metadata.`

**Comment by vladsch** [comment]
Yes but Bridge is still based on NRefactory which has 512 chars limitation
